tlslite/handshakesettings.py

```diff
@@ -239,74 +254,103 @@ def _sanityCheckKeySizes(other):
             raise ValueError("maxKeySize smaller than minKeySize")
 
     @staticmethod
-    def _sanityCheckPrimitivesNames(other):
-        """Check if specified cryptographic primitive names are known"""
-        unknownCiphers = [val for val in other.cipherNames \
-                          if val not in ALL_CIPHER_NAMES]
-        if unknownCiphers:
-            raise ValueError("Unknown cipher name: %s" % unknownCiphers)
-
-        unknownMacs = [val for val in other.macNames \
-                       if val not in ALL_MAC_NAMES]
-        if unknownMacs:
-            raise ValueError("Unknown MAC name: %s" % unknownMacs)
-
-        unknownKex = [val for val in other.keyExchangeNames \
-                      if val not in KEY_EXCHANGE_NAMES]
-        if unknownKex:
-            raise ValueError("Unknown key exchange name: %s" % unknownKex)
-
-        unknownImpl = [val for val in other.cipherImplementations \
-                       if val not in CIPHER_IMPLEMENTATIONS]
+    def _not_matching(values, sieve):
+        """Return list of items from values that are not in sieve."""
+        return [val for val in values if val not in sieve]
+
+    @staticmethod
+    def _sanityCheckCipherSettings(other):
+        """Check if specified cipher settings are known."""
+        not_matching = HandshakeSettings._not_matching
+        unknown = not_matching(other.cipherNames, ALL_CIPHER_NAMES)
+        if unknown:
+            raise ValueError("Unknown cipher name: {0}".format(unknown))
+
+        unknown = not_matching(other.macNames, ALL_MAC_NAMES)
+        if unknown:
+            raise ValueError("Unknown MAC name: {0}".format(unknown))
+
+        unknown = not_matching(other.keyExchangeNames, KEY_EXCHANGE_NAMES)
+        if unknown:
```

Comment:
Nit: You inconsistently renamed the unknownXyz variables here but the other ones kept the original name. Not sure if you want to change that.

Reply:
yeah, it's probably better to be consistent